serialization/src/main/scala/sbt/serialization/CanToString.scala

```diff
@@ -23,6 +35,17 @@ object CanToString {
     _.toString, {
       s: String => TypeExpression.parse(s)._1
     })
```

Comment:
ugh, but we have the picklers in a bogus CanToStringPickler trait - it's just implementation detail that we generate the picklers from CanToString. I think we should 1) move the file/uri/typeexpression picklers to a logical trait, not an implementation-detail-based trait, 2) make the canToString instances private implementation details of the picklers.